tests/unit/processor/hyperscan_resolver/test_hyperscan_resolver.py

```diff
@@ -0,0 +1,769 @@
+                    {
+                        ".*HELLO\\d": "Greeting",
+                        ".*HELL.\\d": "Greeting2",
+                        ".*HEL..\\d": "Greeting3",
+                    }
+                ),
+            },
+        }
+
+        self._load_specific_rule(rule)
+
+        expected = {"to": {"resolve": "something HELLO1"}, "re": {"solved": "Greeting"}}
+        document = {"to": {"resolve": "something HELLO1"}}
+
+        self.object.process(document)
+
+        assert document == expected
+
+
+class TestHyperscanResolverProcessorWithPatterns(BaseProcessorTestCase):
+
+    factory = HyperscanResolverFactory
+
+    CONFIG = {
+        "type": "hyperscan_resolver",
+        "specific_rules": ["tests/testdata/unit/hyperscan_resolver/rules/specific/"],
+        "generic_rules": ["tests/testdata/unit/hyperscan_resolver/rules/generic/"],
+        "tree_config": "tests/testdata/unit/shared_data/tree_config.json",
+    }
+
+    @property
+    def specific_rules_dirs(self):
+        """Return the paths of the specific rules"""
+        return self.CONFIG["specific_rules"]
+
+    @property
+    def generic_rules_dirs(self):
+        """Return the paths of the generic rules"""
+        return self.CONFIG["generic_rules"]
+
+    def _load_specific_rule(self, rule):
+        specific_rule = HyperscanResolverRule._create_from_dict(rule)
+        specific_rule.file_name = "some_file_name_is_required_for_hyperscan"
+        self.object._specific_tree.add_rule(specific_rule, self.logger)
+
+    def test_resolve_dotted_no_conflict_from_file(self):
+        rule = {
+            "filter": "to_resolve",
+            "hyperscan_resolver": {
+                "field_mapping": {"to_resolve": "resolved"},
+                "resolve_from_file": {
+                    "path": "tests/testdata/unit/hyperscan_resolver"
+                    "/resolve_mapping_without_patterns.yml",
+                    "pattern": r"\d*(?P<mapping>[a-z]+)\d*",
+                },
+                "resolve_list": {"FOO": "BAR"},
+            },
+        }
+
+        self._load_specific_rule(rule)
+
+        expected = {"to_resolve": "ab", "resolved": "ab_server_type"}
+        document = {"to_resolve": "ab"}
+
+        self.object.process(document)
+
+        assert document == expected
+
+    def test_resolve_dotted_no_conflict_from_file_and_unbalanced_parenthesis(
+        self,
+    ):
+        rule = {
+            "filter": "to_resolve",
+            "hyperscan_resolver": {
+                "field_mapping": {"to_resolve": "resolved"},
+                "resolve_from_file": {
+                    "path": "tests/testdata/unit/hyperscan_resolver"
+                    "/resolve_mapping_without_patterns.yml",
+                    "pattern": r"\d*(?P<mapping>[a-z]+)c)\d*",
+                },
+            },
+        }
+
+        with pytest.raises(Exception, match="unbalanced parenthesis"):
+            self._load_specific_rule(rule)
```

Comment:
* the name is misleading, because we do not use an event here.
* this is not a processor test because we test the rule
* test on python raises should be the job of the python developers. ;-) but we could raise our own Exceptions if it makes sense

Reply:
I have removed it from the processor tests and added some rule tests instead.